### DISCUSSION TITLE
How Is Gateway Supposed to Work?

### DISCUSSION THREAD

**Discussion by prismalogic** [Eventuous/eventuous#186] [Q&A] [answered]
Hi,

We are looking to use Eventuous in production in order to reduce boilerplate code (thanks @alexeyzimarev for creating this!).

We're playing around with the sample app. Everything makes sense but we just can't make heads or tails of how to get the gateway working.

Here's what we've done so far:

- Removed the RecordPayment endpoint from Bookings/CommandApi.
- Added RecordPayment endpoint to Payments/CommandApi.
- Have the existing Gateway in the Bookings.Payments project.
- Created a subscription to listen to IntegrationSubscription in Bookings project.

The subscription works fine if we look for PaymentEvents.PaymentRecorded event (changing this in the handler). But it does not have any knowledge of IntegrationEvents.BookingPaymentRecorded.

The event gets created successfully but the Transform method never hits. Pretty sure we're missing some piping but haven't been able to figure out where.

We've looked through the discussion, there was one mention of Gateway, but the link is broken.

Would appreciate some help. If there is an example with this working as expected, we'd really appreciate it.

**Comment by alexeyzimarev** [comment, marked as answer]
Thanks for the feedback. I am wondering why you needed to do all that work as it's already in the sample app. The RegisterPayment endpoint in the Bookings app is for easy demo purpose only, but the flow should work as you described if you just run Bookings and Payments without any changes.

Here's the API endpoint in Bookings.Payments: https://github.com/Eventuous/dotnet-sample/blob/baf0d7c8e528afc8a50c53da2417bee8d11a8b8f/Bookings.Payments/Application/CommandApi.cs

Here's the integration subscription in Bookings: https://github.com/Eventuous/dotnet-sample/blob/baf0d7c8e528afc8a50c53da2417bee8d11a8b8f/Bookings/Integration/Payments.cs

**Reply by prismalogic** [reply]
Hi Alexey,

Not sure what to say other than that looks like we were just overcomplicating things in our head where they are extremely simple. Downloaded the sample again and ran it without changing anything. Works beautifully.

Thank you!
